pkg/executables/dockercontainer.go

```diff
@@ -14,6 +14,7 @@ import (
 )
 
 type DockerClient interface {
+	Login(ctx context.Context, endpoint, username, password string) error
```

Comment:
What's the consumer of this interface?

Reply:
it's used when we do the docker login in factory.go:

`err := f.executablesConfig.dockerClient.Login(context.Background(), f.registryMirror, username, password)`